consensus/consortium/v2/consortium.go

```diff
@@ -1620,3 +1641,34 @@ func (c *Consortium) IsPeriodBlock(chain consensus.ChainHeaderReader, header *ty
 	}
 	return uint64(header.Time/dateInSeconds) > snap.CurrentPeriod
 }
+
+// IsTrippEffective returns if the finality vote rule change is effective. This change is effective
+// after the period of Tripp block not right after the Tripp block.
+func (c *Consortium) IsTrippEffective(chain consensus.ChainHeaderReader, header *types.Header) bool {
+	if c.chainConfig.IsTripp(header.Number) {
+		if c.testTrippEffective {
+			return true
+		}
+
+		snap, err := c.snapshot(chain, header.Number.Uint64()-1, header.ParentHash, nil)
+		if err != nil {
+			log.Error("Failed to get snapshot", "err", err)
+			parent := c.getLastCheckpointHeader(chain, header)
+			if parent == nil {
+				return false
+			}
+
+			return uint64(header.Time/dateInSeconds) > uint64(parent.Time/dateInSeconds)
+		}
+
+		if snap.CurrentPeriod > c.chainConfig.TrippPeriod.Uint64() {
+			return true
+		}
+
+		if c.IsPeriodBlock(chain, header) {
```

Comment:
I think we can check like `header.Number%EpochV2==0 && header.Time/dateInSeconds>c.chainConfig.TrippPeriod`, this can reduce a `snapshot` in IsPeriodBlock

Reply:
fixed